analyzers/src/SonarAnalyzer.SourceGenerator/RuleCatalogGenerator.cs

```diff
@@ -0,0 +1,106 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Newtonsoft.Json.Linq;
+
+namespace SonarAnalyzer.SourceGenerator
+{
+    [Generator]
+    [ExcludeFromCodeCoverage]
+    public class RuleCatalogGenerator : ISourceGenerator
+    {
+        private const string SonarWayFileName = "Sonar_way_profile.json";
+
+        public void Initialize(GeneratorInitializationContext context)
+        {
+            // Not needed
+        }
+
+        public void Execute(GeneratorExecutionContext context)
+        {
+            if (!context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.projectdir", out var projectDir))
+            {
+                throw new NotSupportedException("Cannot find ProjectDir");
+            }
+            var language = Path.GetFileName(projectDir.TrimEnd(Path.DirectorySeparatorChar)) switch
+            {
+                "SonarAnalyzer.CSharp" => "cs",
+                "SonarAnalyzer.VisualBasic" => "vbnet",
+                _ => throw new ArgumentException($"Unexpected projectDir: {projectDir}")
+            };
+            context.AddSource($"RuleCatalog.{language}.g.cs", GenerateSource(RuleDescriptorArguments(Path.Combine(projectDir, "..", "..", "rspec", language))));
+        }
+
+        private static IEnumerable<string[]> RuleDescriptorArguments(string rspecDirectory)
+        {
+            var sonarWay = new HashSet<string>(JObject.Parse(File.ReadAllText(Path.Combine(rspecDirectory, SonarWayFileName)))["ruleKeys"].Values<string>());
+            foreach (var jsonPath in Directory.GetFiles(rspecDirectory, "*.json").Where(x => Path.GetFileName(x) != SonarWayFileName))
+            {
+                var json = JObject.Parse(File.ReadAllText(jsonPath));
+                var id = Path.GetFileName(jsonPath).Split('_').First();
+                yield return new[]
+                {
+                    Encode(id),
+                    Encode(json.Value<string>("title")),
+                    Encode(json.Value<string>("type")),
+                    Encode(json.Value<string>("defaultSeverity")),
+                    $"SourceScope.{json.Value<string>("scope")}",
+                    sonarWay.Contains(id).ToString().ToLower(),
+                    Encode(File.ReadAllText(Path.ChangeExtension(jsonPath, ".html")))
```

Comment:
FIXME: I have a feeling that we'll need only 1st paragraph or something.

Reply:
The feeling was correct. I need to do this:
https://github.com/SonarSource/sonar-dotnet/blob/14faf2a68ad2b157c76c490a1b806152f2d94898/scripts/rspec/rspec2resx.ps1#L109-L118

Reply:
The `\\!` and `\\}` parts are not needed anymore (JIRA messencoding)